workshops/models.py

```diff
@@ -397,6 +400,42 @@ def is_staff(self):
     def is_admin(self):
         return is_admin(self)
 
+    @property
+    def missing_swc_instructor_requirements(self):
+        """ Returns set of requirements that are not passed yet by the
+        trainee and are mandatory to become SWC Instructor.. """
+        training = TrainingRequirement.objects.get(name='Training')
+        homework = TrainingRequirement.objects.get(name='SWC Homework')
+        discussion = TrainingRequirement.objects.get(name='Discussion')
+        demo = TrainingRequirement.objects.get(name='SWC Demo')
+
+        requirements = {training, homework, discussion, demo}
+        progresses = self.trainingprogress_set.filter(discarded=False, state='p')
+        passed_requirements = {p.requirement for p in progresses}
+        return requirements - passed_requirements
+
+    @property
+    def missing_dc_instructor_requirements(self):
+        """ Returns set of requirements that are not passed yet by the
+        trainee and are mandatory to become DC Instructor.. """
+        training = TrainingRequirement.objects.get(name='Training')
+        homework = TrainingRequirement.objects.get(name='DC Homework')
+        discussion = TrainingRequirement.objects.get(name='Discussion')
+        demo = TrainingRequirement.objects.get(name='DC Demo')
+
+        requirements = {training, homework, discussion, demo}
+        progresses = self.trainingprogress_set.filter(discarded=False, state='p')
+        passed_requirements = {p.requirement for p in progresses}
+        return requirements - passed_requirements
+
+    @property
+    def training_tasks(self):
+        """ Returns Tasks related to Instuctor Training events at which this
+        person was trained."""
+        return Task.objects.filter(person=self,
+                                   role__name='learner',
+                                   event__tags__name='TTT')
```

Comment:
This property is used in templates: 

- https://github.com/swcarpentry/amy/pull/871/files#diff-f4cb88d18f36cf5a6e5fc2aeb1a32602R68 
- https://github.com/swcarpentry/amy/pull/871/files#diff-e4b786fea3b24ff03fcc3d341a1c213bR53